runtime/src/bank.rs

```diff
@@ -6642,6 +6649,23 @@ impl Bank {
 
         total_accounts_stats
     }
+
+    /// Set the Cost Tracker's account data size limit based on the current accounts data size.
+    ///
+    /// NOTE: Use `&mut` for `self` to ensure this fn is only called in constructors.
+    fn set_cost_tracker_account_data_size_limit(&mut self, accounts_data_len: u64) {
+        if !self
+            .feature_set
+            .is_active(&feature_set::cap_accounts_data_len::id())
+        {
+            return;
+        }
+
+        let remaining_accounts_data_len = MAX_ACCOUNTS_DATA_LEN.saturating_sub(accounts_data_len);
+        self.write_cost_tracker()
+            .unwrap()
+            .set_account_data_size_limit(remaining_accounts_data_len);
+    }
```

Comment:
I would replace this function with something like 
```
CostTracker::new( 
    if feature_enabled { Some(remaining_value) }
    else {None},
    ... ... 
)
```
I prefer `limit` as a const to be set at construction at once.

Reply:
Ok, done!